rio_tiler/mosaic/reader.py

```diff
@@ -41,6 +43,76 @@ def _filter_tasks(tasks: TaskType):
             pass
 
 
+@dataclass
+class MosaicReader:
+    """
+    Mosaic Reader.
+
+    Examples
+    --------
+    with MosaicReader(assets, reader=COGReader) as mosaic:
+        mosaic.tile(...)
+
+    with MosaicReader(assets, reader=STACReader) as mosaic:
+        mosaic.tile(...)
+
+    Attributes
+    ----------
+    assets: str
+        Cloud Optimized GeoTIFF path.
+    reader: rio_tiler.io.BaseReader, optional
+        Rio-tiler reader class.
+
+    Methods
+    -------
+    tile(0, 0, 0, i**kwargs)
+        Return map tile from multiple observation.
+    part((0,10,0,10), **kwargs)
+        Return array for a specific bbox from multiple observation.
+    preview(**kwargs)
+        Return a preview from multiple observation.
+
+    """
+
+    assets: Sequence[str]
+    reader: BaseReader
```

Comment:
not sure if we want to make COGReader the default here?

Reply:
I think `BaseReader` is good here